tests/func/test_scm.py

```diff
@@ -34,27 +34,27 @@ def test_init_sub_dir(tmp_dir):
     assert scm.root_dir == os.fspath(tmp_dir)
 
 
-class TestSCMGit(TestGit):
-    def test_commit(self):
-        G = Git(self._root_dir)
-        G.add(["foo"])
-        G.commit("add")
-        self.assertIn("foo", self.git.git.ls_files())
-
-    def test_is_tracked(self):
-        foo = os.path.abspath(self.FOO)
-        G = Git(self._root_dir)
-        G.add([self.FOO, self.UNICODE])
-        self.assertTrue(G.is_tracked(foo))
-        self.assertTrue(G.is_tracked(self.FOO))
-        self.assertTrue(G.is_tracked(self.UNICODE))
-        G.commit("add")
-        self.assertTrue(G.is_tracked(foo))
-        self.assertTrue(G.is_tracked(self.FOO))
-        G.gitpython.repo.index.remove([self.FOO], working_tree=True)
-        self.assertFalse(G.is_tracked(foo))
-        self.assertFalse(G.is_tracked(self.FOO))
-        self.assertFalse(G.is_tracked("not-existing-file"))
+def test_commit(tmp_dir, scm):
+    tmp_dir.gen({"foo": "foo"})
+    scm.add(["foo"])
+    scm.commit("add")
+    assert "foo" in Repo(tmp_dir).git.ls_files()
```

Comment:
How about this one.
```
scm.is_tracked("foo")
```

instead of 

```
    assert "foo" in Repo(tmp_dir).git.ls_files()
```

Reply:
I've replaced `ls_files` with `is_tracked`. But I'm not sure if `test_commit` is really needed because it seems that `test_is_tracked` fully covers `test_commit` case.

Reply:
So, the reason that this did not use `is_tracked` before is that `is_tracked` is still using our SCM class, and the tests for `is_tracked` itself also depends on the assumption that `scm.commit` works (so we want to test `scm.commit` independently of `is_tracked`.

To verify that `scm.commit` works, we were testing it directly using CLI git before (the `git.ls_files` call spawns an actual CLI `git ls-files` call via gitpython).

Reply:
ok. Now `test_commit` makes sense.